app/views/spree/admin/orders/_shipment_manifest.html.haml

```diff
@@ -14,7 +14,9 @@
           = "#{count} x #{t(state.humanize.downcase, scope: [:spree, :shipment_states], default: [:missing, "none"])}"
       - unless shipment.shipped?
         %td.item-qty-edit.hidden
-          = number_field_tag :quantity, item.quantity, :min => 0, :class => "line_item_quantity", :size => 5, :max => item.variant.on_hand + item.quantity
+          - html_options = { :min => 0, :class => "line_item_quantity", :size => 5 }
+          - html_options.merge!({ :max => item.variant.on_hand + item.quantity }) unless item.variant.on_demand
+          = number_field_tag :quantity, item.quantity, html_options
```

Comment:
we're stuffing quite some Ruby logic into the view already. I think this better fit into a helper method like `quantity_field_tag` :ok_hand:

Reply:
Good idea! f32dc8c50d2c4b810bb880b3850ebec68bd346cf